tty/pc-tty/ttypc_vtf.c

```diff
@@ -634,212 +656,340 @@ void _ttypc_vtf_sgr(ttypc_virt_t *virt)
+	case 1:  /* GATM - guarded area transfer mode */
+	case 2:  /* KAM - keyboard action mode */
+	case 3:  /* CRM - Control Representation mode */
 		break;
 
-	case 4:		/* IRM - insert replacement mode */
-		virt->m_irm = 0;
+	case 4:  /* IRM - insert replacement mode */
+		vt->irm = 1;
 		break;
 
-	case 5:		/* SRTM - status report transfer mode */
-	case 6:		/* ERM - erasue mode */
-	case 7:		/* VEM - vertical editing mode */
-	case 10:	/* HEM - horizontal editing mode */
-	case 11:	/* PUM - position unit mode */
-	case 12:	/* SRM - send-receive mode */
-	case 13:	/* FEAM - format effector action mode */
-	case 14:	/* FETM - format effector transfer mode */
-	case 15:	/* MATM - multiple area transfer mode */
-	case 16:	/* TTM - transfer termination */
-	case 17:	/* SATM - selected area transfer mode */
-	case 18:	/* TSM - tabulation stop mode */
-	case 19:	/* EBM - editing boundary mode */
-	case 20:	/* LNM - line feed / newline mode */
+	case 5:  /* SRTM - status report transfer mode */
+	case 6:  /* ERM - erasue mode */
+	case 7:  /* VEM - vertical editing mode */
+	case 10: /* HEM - horizontal editing mode */
+	case 11: /* PUM - position unit mode */
+	case 12: /* SRM - send-receive mode */
+	case 13: /* FEAM - format effector action mode */
+	case 14: /* FETM - format effector transfer mode */
+	case 15: /* MATM - multiple area transfer mode */
+	case 16: /* TTM - transfer termination */
+	case 17: /* SATM - selected area transfer mode */
+	case 18: /* TSM - tabulation stop mode */
+	case 19: /* EBM - editing boundary mode */
+		break;
+
+	case 20: /* LNM - line feed / newline mode */
+		vt->lnm = 1;
 		break;
 	}
 }
 
 
-/* request terminal parameters */
-void _ttypc_vtf_reqtparm(ttypc_virt_t *virt)
+void _ttypc_vtf_resetansi(ttypc_vt_t *vt)
 {
-	/* static uint8_t *answr = (uint8_t *)"\033[3;1;1;120;120;1;0x"; */
+	switch (vt->parms[0]) {
+	case 0:  /* error, ignored */
+	case 1:  /* GATM - guarded area transfer mode */
+	case 2:  /* KAM - keyboard action mode */
+	case 3:  /* CRM - Control Representation mode */
+		break;
+
+	case 4:  /* IRM - insert replacement mode */
+		vt->irm = 0;
+		break;
+
+	case 5:  /* SRTM - status report transfer mode */
+	case 6:  /* ERM - erasue mode */
+	case 7:  /* VEM - vertical editing mode */
+	case 10: /* HEM - horizontal editing mode */
+	case 11: /* PUM - position unit mode */
+	case 12: /* SRM - send-receive mode */
+	case 13: /* FEAM - format effector action mode */
+	case 14: /* FETM - format effector transfer mode */
+	case 15: /* MATM - multiple area transfer mode */
+	case 16: /* TTM - transfer termination */
+	case 17: /* SATM - selected area transfer mode */
+	case 18: /* TSM - tabulation stop mode */
+	case 19: /* EBM - editing boundary mode */
+		break;
 
-	/* respond(answr); */
+	case 20: /* LNM - line feed / newline mode */
+		vt->lnm = 0;
+		break;
+	}
+}
+
+
+void _ttypc_vtf_reqtparm(ttypc_vt_t *vt)
+{
+	static char *tparm = "\033[3;1;1;120;120;1;0x";
+	write(0, tparm, sizeof(tparm));
```

Comment:
sizeof(tparm ) = 4 (sizeof(char *)). It should be `static const char tparm [] = "\033[3;1;1;120;120;1;0x";`

Reply:
Fixed.